src/utils/fetch.js

```diff
@@ -93,12 +93,10 @@ export const del = (url, action_success, action_failure, dispatch) => {
     method: 'DELETE',
     credentials: 'same-origin',
   })
-    .then(data =>
-      dispatch({
-        type: action_success.type,
-        id: action_success.id,
-      })
-    )
+    .then(data => {
+      dispatch({ type: action_success.type });
+      dispatch(action_success.update);
```

Comment:
Why are there 2 dispatch here?

Reply:
@mertkahyaoglu  Because the original logic sends two dispatches. For example:
```js
export const deleteDraft = (directory, filename) => dispatch => {
  return fetch(draftAPIUrl(directory, filename), {
    method: 'DELETE',
    credentials: 'same-origin',
  })
    .then(data => {
      dispatch({ type: DELETE_DRAFT_SUCCESS });
      dispatch(fetchDrafts(directory));
    })
    .catch(error =>
      dispatch({
        type: DELETE_DRAFT_FAILURE,
        error,
      })
    );
```

Reply:
Oh, is it because we are not updating state, just refetching all drafts again, right? That's bad actually 😄

Reply:
Hmm.. this double dispatch could also be the reason behind the race condition issues with the frontend layer and the Ruby backend.. not sure though.

Reply:
Fixed this double-dispatch in https://github.com/jekyll/jekyll-admin/pull/578/commits/b11903bf670549a540cd38faa6281b76a2f4482e